File: split/newwritewhatsit.pas
procedure newwritewhatsit(w: smallnumber);
begin
  newwhatsit(curchr, w);
  if w <> 2 then
    scanfourbitint
  else
  begin
    scanint;
    if curval < 0 then
      curval := 17
    else if curval > 15 then
      curval := 16;
  end;
  mem[curlist.tailfield + 1].hh.lh := curval;
end;

File: cpp/newwritewhatsit.cpp
#include "newwritewhatsit.h"
#include "newwhatsit.h"
#include "scanfourbitint.h"
#include "scanint.h"


void newwritewhatsit(smallnumber w)
{
	newwhatsit(curchr, w);
	if (w != write_node_size)
		scanfourbitint();
	else
	{
		scanint();
		if (curval < 0)
			curval = 17;
		else 
			if (curval > 15)
				curval = 16;
	}
	write_stream(tail) = curval;
}
